airbyte-api/src/main/openapi/config.yaml

```diff
@@ -906,6 +906,27 @@ paths:
           description: Connection not found
         "422":
           $ref: "#/components/responses/InvalidInput"
+  /v1/operations/check:
```

Comment:
I'm going to assume there's some use case for this that I don't know about, It's not clear to me why we need _both_  `/v1/operations/check` and `/v1/operations/check_for_update`. Can't we always just create an `OperationCreate` object from either an `OperationCreate` or an `OperationUpdate`? The only reason I can think of needing the update version is if there is some limited number of transformations that are allowed on an existing operation that don't apply when you're creating one from scratch.

Reply:
Maybe you used this as a pattern?

```
  /v1/sources/check_connection:
    post:
      tags:
        - source
      summary: Check connection to the source
      operationId: checkConnectionToSource
      requestBody:
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/SourceIdRequestBody"
        required: true
      responses:
        "200":
          description: Successful operation
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/CheckConnectionRead"
        "404":
          description: Source not found
        "422":
          $ref: "#/components/responses/InvalidInput"
  /v1/sources/check_connection_for_update:
    post:
      tags:
        - source
      summary: Check connection for a proposed update to a source
      operationId: checkConnectionToSourceForUpdate
      requestBody:
        content:
          application/json:
            schema:
              $ref: "#/components/schemas/SourceUpdate"
        required: true
      responses:
        "200":
          description: Successful operation
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/CheckConnectionRead"
        "404":
          description: Source not found
        "422":
          $ref: "#/components/responses/InvalidInput"
```

if so, this case is a bit different. the case being modeled in the source case is when the caller doesn't have access to the whole source object just the id but in the latter case is trying to check a source that doesn't exist yet. i'm not even sure this pattern makes too much sense in the source case, i think it might have just happened because the one that takes just a source id happened first. we could probably just say you need to send the whole object every time.

Reply:
The check route is to be used from the UI as artem was mentioning a need to validate before actually creating an operation

For now, it's really simple, but we could extend the check to validate further the parameters of the operation in the future.

For example if you provide a git_url, checking if it's valid and we're able to pull/clone from it etc